hal/armv7m/imxrt/117x/imxrt117x.c

```diff
@@ -837,6 +851,15 @@ int hal_platformctl(void *ptr)
 			}
 			break;
 
+		case pctl_resetSlice:
+			if (data->action == pctl_set) {
+				if (data->resetSlice.index >= 0 && data->resetSlice.index <= 11) {
```

Comment:
After adding enums with indexes (see previous comment) it may be:
(just an example, depends on the  enum names you choose)
```suggestion
				if ((data->resetSlice.index >= pctl_resetSliceMega) && (data->resetSlice.index <= pctl_resetSliceCM7Mem)) {
```